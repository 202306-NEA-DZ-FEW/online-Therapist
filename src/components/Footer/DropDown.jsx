import Link from "next/link";
import { useTranslation } from "next-i18next";
import { useState } from "react";

const DropDown = () => {
    const [open, setOpen] = useState(false);
    const { t } = useTranslation("common");

    return (
        <div className='relative'>
            <a
                className='hover:underline cursor-pointer mb-3'
                onClick={() => setOpen(!open)}
            >
                {t("footer.about")}
            </a>
            <div className={`${open ? "block" : "hidden"} absolute`}>
                <div className='ml-4 '>
                    <Link
                        href='/about-us/'
                        className='mr-4 hover:underline md:mr-6 '
                    >
                        {t("footer.aboutUs")}
                    </Link>
                </div>
                <div className='ml-4 text-dark'>
                    <Link
                        href='/team/'
                        className='mr-4 hover:underline md:mr-6 '
                    >
                        {t("footer.team")}
                    </Link>
                </div>
            </div>
        </div>
    );
};

export default DropDown;
