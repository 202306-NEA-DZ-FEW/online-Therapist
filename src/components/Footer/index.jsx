import DropDown from "./DropDown";
export default function Footer() {
    return (
        <footer className='bg-DarkTeal dark:bg-gray-900'>
            <div className='mx-auto w-full max-w-screen-xl p-4 py-6 lg:py-8'>
                <div className='md:flex md:justify-between'>
                    <div className='mb-6 md:mb-0 '>
                        <h2 className='text-white text-2xl text-center mr-4'>
                            Subscribe
                        </h2>
                        <p className='text-dark mt-3 mb-5'>
                            We&apos;ll never to spam you or share your email
                        </p>
                        <div className='flex items-center flex-shrink-0 w-full mx-auto sm:w-auto'>
                            <form
                                action='#'
                                className='flex flex-col items-center w-full md:flex-row'
                            >
                                <input
                                    type='email'
                                    id='email'
                                    placeholder='Enter your email'
                                    class='bg-white border border-gray-300 text-gray-900 md:w-64 mb-2 md:mb-0 md:mr-4 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-600 dark:border-gray-500 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500'
                                    required
                                />
                                <button
                                    type='submit'
                                    className='text-white bg-Teal hover:bg-white hover:text-Teal focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm w-full sm:w-auto px-5 py-2.5 center '
                                >
                                    SUBSCRIBE
                                </button>
                            </form>
                        </div>
                    </div>
                    <div>
                        <ul className='flex flex-wrap items-center mt-3 text-sm font-medium text-white dark:text-gray-400 sm:mt-0 mb-10 '>
                            <li>
                                <a
                                    href='#'
                                    className='mr-4 hover:underline md:mr-6 '
                                >
                                    Home
                                </a>
                            </li>
                            <li>
                                <a
                                    href='#'
                                    className='mr-4 hover:underline md:mr-6 pl-8'
                                >
                                    Blogs
                                </a>
                            </li>
                            <li>
                                <DropDown />
                            </li>
                            <li>
                                <a href='#' className='hover:underline'>
                                    Contact
                                </a>
                            </li>
                        </ul>
                        <div className='flex mt-10 space-x-5 sm:justify-center item-center sm:mt-0 pt-5'>
                            <a
                                href='https://www.facebook.com/recodedofficial'
                                className='text-dark hover:text-white dark:hover:text-white'
                            >
                                <svg
                                    className='w-4 h-4'
                                    aria-hidden='true'
                                    xmlns='http://www.w3.org/2000/svg'
                                    fill='currentColor'
                                    viewBox='0 0 8 19'
                                >
                                    <path
                                        fill-rule='evenodd'
                                        d='M6.135 3H8V0H6.135a4.147 4.147 0 0 0-4.142 4.142V6H0v3h2v9.938h3V9h2.021l.592-3H5V3.591A.6.6 0 0 1 5.592 3h.543Z'
                                        clip-rule='evenodd'
                                    />
                                </svg>
                                <span className='sr-only'>Facebook page</span>
                            </a>

                            <a
                                href='https://twitter.com/recodedofficial'
                                className='text-dark hover:text-white dark:hover:text-white'
                            >
                                <svg
                                    className='w-4 h-4'
                                    aria-hidden='true'
                                    xmlns='http://www.w3.org/2000/svg'
                                    fill='currentColor'
                                    viewBox='0 0 20 17'
                                >
                                    <path
                                        fill-rule='evenodd'
                                        d='M20 1.892a8.178 8.178 0 0 1-2.355.635 4.074 4.074 0 0 0 1.8-2.235 8.344 8.344 0 0 1-2.605.98A4.13 4.13 0 0 0 13.85 0a4.068 4.068 0 0 0-4.1 4.038 4 4 0 0 0 .105.919A11.705 11.705 0 0 1 1.4.734a4.006 4.006 0 0 0 1.268 5.392 4.165 4.165 0 0 1-1.859-.5v.05A4.057 4.057 0 0 0 4.1 9.635a4.19 4.19 0 0 1-1.856.07 4.108 4.108 0 0 0 3.831 2.807A8.36 8.36 0 0 1 0 14.184 11.732 11.732 0 0 0 6.291 16 11.502 11.502 0 0 0 17.964 4.5c0-.177 0-.35-.012-.523A8.143 8.143 0 0 0 20 1.892Z'
                                        clip-rule='evenodd'
                                    />
                                </svg>
                                <span className='sr-only'>Twitter page</span>
                            </a>
                            <a
                                href='https://github.com/202306-NEA-DZ-FEW/online-Therapist'
                                className='text-dark hover:text-white dark:hover:text-white'
                            >
                                <svg
                                    className='w-4 h-4'
                                    aria-hidden='true'
                                    xmlns='http://www.w3.org/2000/svg'
                                    fill='currentColor'
                                    viewBox='0 0 20 20'
                                >
                                    <path
                                        fill-rule='evenodd'
                                        d='M10 .333A9.911 9.911 0 0 0 6.866 19.65c.5.092.678-.215.678-.477 0-.237-.01-1.017-.014-1.845-2.757.6-3.338-1.169-3.338-1.169a2.627 2.627 0 0 0-1.1-1.451c-.9-.615.07-.6.07-.6a2.084 2.084 0 0 1 1.518 1.021 2.11 2.11 0 0 0 2.884.823c.044-.503.268-.973.63-1.325-2.2-.25-4.516-1.1-4.516-4.9A3.832 3.832 0 0 1 4.7 7.068a3.56 3.56 0 0 1 .095-2.623s.832-.266 2.726 1.016a9.409 9.409 0 0 1 4.962 0c1.89-1.282 2.717-1.016 2.717-1.016.366.83.402 1.768.1 2.623a3.827 3.827 0 0 1 1.02 2.659c0 3.807-2.319 4.644-4.525 4.889a2.366 2.366 0 0 1 .673 1.834c0 1.326-.012 2.394-.012 2.72 0 .263.18.572.681.475A9.911 9.911 0 0 0 10 .333Z'
                                        clip-rule='evenodd'
                                    />
                                </svg>
                                <span className='sr-only'>GitHub account</span>
                            </a>
                        </div>
                    </div>
                </div>
                <hr className='my-6 border-gray-200 sm:mx-auto dark:border-gray-700 lg:my-8' />
                <div className='sm:flex sm:items-center sm:justify-center'>
                    <span className='text-sm text-dark sm:text-center hover:text-white'>
                        © 2023{" "}
                        <a href='#' className='hover:underline'>
                            InnerSpace™
                        </a>
                        . All Rights Reserved.
                    </span>
                </div>
            </div>
        </footer>
    );
}
