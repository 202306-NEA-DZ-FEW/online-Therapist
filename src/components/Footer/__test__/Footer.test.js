import renderer from "react-test-renderer";
import { useRouter } from "next/router";
import Footer from "..";

// Mock the useRouter function
jest.mock("next/router", () => ({
    useRouter: jest.fn(),
}));

it("renders correctly", () => {
    // Configure the mock useRouter function to return a locale
    useRouter.mockImplementation(() => ({
        locale: "en",
    }));

    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
});
