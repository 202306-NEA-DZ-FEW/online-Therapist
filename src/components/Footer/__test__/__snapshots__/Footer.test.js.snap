// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="bg-DarkTeal"
>
  <div
    className="mx-auto w-full max-w-screen-xl p-10 xl:p-2 py-8"
  >
    <div
      className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 px-4"
    >
      <div
        className="md:col-span-2 lg:col-span-1 flex flex-col items-center justify-center"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height={125}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2FImages%2Flogo2.png&w=256&q=75"
            srcSet="/_next/image?url=%2FImages%2Flogo2.png&w=128&q=75 1x, /_next/image?url=%2FImages%2Flogo2.png&w=256&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={125}
          />
        </a>
        <p
          className="text-white mt-3 mb-5 rtl:mr-1 text-center ltr:md:text-left rtl:md:text-right font-medium"
        >
          footer.motto
        </p>
      </div>
      <div>
        <h2
          className="text-white text-2xl my-3 text-center ltr:md:text-left rtl:md:text-right rtl:ml-4"
        >
          footer.pages
        </h2>
        <hr
          className="ltr:md:mr-5 rtl:md:ml-5 mb-5"
        />
        <ul
          className="flex md:flex-col justify-evenly gap-4 mt-3 text-sm font-medium text-white sm:mt-0 mb-10 rtl:text-right h-16 md:h-auto"
        >
          <li>
            <a
              className="hover:underline ltr:md:mr-6"
              href="/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              footer.home
            </a>
          </li>
          <li>
            <a
              className="hover:underline ltr:md:mr-6 "
              href="/blogs"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              footer.blogs
            </a>
          </li>
          <li>
            <a
              className="hover:underline"
              href="/contact"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              footer.contact
            </a>
          </li>
          <li>
            <div
              className="relative"
            >
              <a
                className="hover:underline cursor-pointer mb-3"
                onClick={[Function]}
              >
                footer.about
              </a>
              <div
                className="hidden absolute"
              >
                <div
                  className="ml-4 "
                >
                  <a
                    className="mr-4 hover:underline md:mr-6 "
                    href="/about-us"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onTouchStart={[Function]}
                  >
                    footer.aboutUs
                  </a>
                </div>
                <div
                  className="ml-4 text-dark"
                >
                  <a
                    className="mr-4 hover:underline md:mr-6 "
                    href="/team"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onTouchStart={[Function]}
                  >
                    footer.team
                  </a>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
      <div
        className="mb-6 md:mb-0 rtl:text-right"
      >
        <h2
          className="text-white text-2xl my-3 text-center ltr:md:text-left rtl:md:text-right rtl:ml-4"
        >
          footer.subscribe
        </h2>
        <hr
          className="ltr:md:mr-5 rtl:md:ml-5 mb-5"
        />
        <p
          className="text-white mt-3 mb-5 rtl:mr-1 text-center ltr:md:text-left rtl:md:text-right"
        >
          footer.spam
        </p>
        <div
          className="flex items-center flex-shrink-0 w-full mx-auto sm:w-auto rtl:ml-4 ltr:md:mr-4"
        >
          <form
            action="#"
            className="flex flex-col items-center w-full md:flex-row rtl:text-right"
          >
            <input
              className="bg-white border border-gray-300 md:w-64 mb-2 md:mb-0 md:mr-2 text-sm rounded-lg block w-full p-2.5 focus:outline-none rtl:m-3"
              id="email"
              placeholder="footer.email"
              required={true}
              type="email"
            />
            <button
              className="text-white bg-Teal hover:bg-white hover:text-Teal focus:ring-4 focus:outline-none  font-medium rounded-lg text-sm w-full sm:w-auto px-5 py-2.5 center "
              type="submit"
            >
              footer.subscribe
            </button>
          </form>
        </div>
      </div>
    </div>
    <div
      class="inline-flex items-center justify-center w-full"
    >
      <hr
        class="w-full h-px my-8 bg-white border-0"
      />
      <span
        class="absolute px-3 -translate-x-1/2 bg-DarkTeal left-1/2"
      >
        <div
          className="flex space-x-3 justify-center items-center sm:mt-0 rtl:flex-row-reverse"
        >
          <a
            className="text-white hover:text-black text-center rtl:ml-4"
            href="https://www.facebook.com/recodedofficial"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              fill="currentColor"
              height={23}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width={23}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
              />
            </svg>
            <span
              className="sr-only"
            >
              Facebook page
            </span>
          </a>
          <a
            className="text-white hover:text-black text-center rtl:ml-2"
            href="https://twitter.com/recodedofficial"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              fill="currentColor"
              height={23}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width={23}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.6.75h2.454l-5.36 6.142L16 15.25h-4.937l-3.867-5.07-4.425 5.07H.316l5.733-6.57L0 .75h5.063l3.495 4.633L12.601.75Zm-.86 13.028h1.36L4.323 2.145H2.865l8.875 11.633Z"
              />
            </svg>
            <span
              className="sr-only"
            >
              Twitter page
            </span>
          </a>
          <a
            className="text-white hover:text-black text-center rtl:ml-2"
            href="https://github.com/202306-NEA-DZ-FEW/online-Therapist"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <svg
              fill="currentColor"
              height={23}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 496 512"
              width={23}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
            <span
              className="sr-only"
            >
              GitHub account
            </span>
          </a>
        </div>
      </span>
    </div>
    <div
      className="flex items-center justify-center rtl:flex-row-reverse"
    >
      <span
        className="text-sm text-dark text-center sm:text-left hover:text-white rtl:ml-4"
      >
        © 2023
         
        <a
          className="hover:underline"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          footer.innerSpace
        </a>
        footer.copyright
      </span>
    </div>
  </div>
</footer>
`;
